admin/init.sls

```diff
@@ -22,22 +22,37 @@ UTC:
     - mode: 644
     - source: salt://{{ tpldir }}/files/hosts
 
-sshkeys-dir:
-  file.directory:
-    - name: {{ root.home }}/.ssh
-    - user: {{ root.user }}
-    - group: {{ root.group }}
-    - mode: 700
-
-sshkeys:
+/etc/sshd_config:
   file.managed:
-    - name: {{ root.home }}/.ssh/authorized_keys
-    - user: {{ root.user }}
-    - group: {{ root.group }}
-    - mode: 600
-    - contents:
-      {% for ssh_user in admin.ssh_users %}
-      - {% include tpldir ~ '/files/ssh/' ~ ssh_user ~ '.pub' %}
-      {% endfor %}
-    - require:
-      - file: sshkeys-dir
+    - user: root
+    {% if grains['os'] == 'MacOS' %}
+    - group: wheel
+    {% elif grains['os'] == 'Ubuntu' %}
+    - group: root
+    {% endif %}
+    - mode: 644
+    - source: salt://{{ tpldir }}/files/sshd_config
+
+wheel:
+    group.present
+
+{% for ssh_user in admin.ssh_users %}
+{{ ssh_user }}:
+    group.present:
+        - members:
+            - {{ ssh_user }}
+    user.present:
+        - empty_password: True
+        - gid_from_name: True
+        - groups:
+            - wheel
+        - require:
+            - group:
+                - wheel
+                - {{ ssh_user }}
+    ssh_auth.present:
```

Comment:
We'll need to keep the two-state solution of `file.managed` to continue being able to handle SSH key rotation and revocation.